packages/dds/tree/src/core/schema-stored/schema.ts

```diff
@@ -119,6 +132,13 @@ export interface TreeFieldStoredSchema {
 	 * If not specified, types are unconstrained.
 	 */
 	readonly types: TreeTypeSet;
+
+	/**
+	 * Portion of the metadata which can be persisted.
+	 * @remarks
+	 * Discarded when encoding to {@link SchemaFormatVersion.V1}.
+	 */
+	persistedMetadata?: PersistedMetadataFormat;
```

Comment:
Cases which discard metadata are important (thanks for documenting the main one in the doc comment).

We can make such discarding of metadata more explicit if we mark this as a require field. This will also help find any places which need to be updated to preserve metadata.
```suggestion
	persistedMetadata: PersistedMetadataFormat | undefined;
```